iml-gui/crate/src/components/command_modal.rs

```diff
@@ -226,45 +313,219 @@ fn command_item_view(x: &Command, open: bool) -> Node<Msg> {
+    let icon = step_status_icon(step, is_open);
+    let item_caption = div![
+        class![C.flex],
+        div![
+            attrs![At::Style => "flex: 0 0 1em"],
+            class![C.mx_2, C.cursor_pointer],
+            icon,
+            simple_ev(Ev::Click, Msg::Click(TypedId::Step(step.id))),
+        ],
+        div![
+            class![C.flex_grow, C.cursor_pointer, C.underline],
+            step.class_name,
+            simple_ev(Ev::Click, Msg::Click(TypedId::Step(step.id))),
+        ],
+    ];
+    let item_body = if !is_open {
+        empty!()
+    } else {
+        // note, we cannot just use the Debug instance for step.args,
+        // because the keys order changes every time the HashMap is traversed
+        let mut arg_keys = step.args.keys().collect::<Vec<&String>>();
+        arg_keys.sort();
+        let mut arg_str = String::with_capacity(step.args.len() * 10);
+        for k in arg_keys {
+            arg_str.push_str(k);
+            arg_str.push_str(": ");
+            arg_str.push_str(&format!(
+                "{}\n",
+                step.args.get(k).unwrap_or(&serde_json::value::Value::Null)
+            ));
+        }
+        let pre_class = class![
+            C.p_2, C.m_2
+            C.leading_tight,
+            C.text_gray_100,
+            C.bg_gray_900,
+            C.overflow_x_hidden,
+            C.whitespace_pre_line,
+            C.break_all,
+        ];
+        div![
+            class![C.flex],
+            div![
+                attrs![At::Style => "flex: 0 0 1em"],
+                class![C.border_r_2, C.border_gray_300, C.hover__border_gray_600],
+                simple_ev(Ev::Click, Msg::Click(TypedId::Step(step.id))),
+            ],
+            div![attrs![At::Style => "flex: 0 0 1em"],],
+            div![
+                class![C.float_right, C.flex_grow],
+                h4![class![C.text_lg, C.font_medium], "Arguments"],
+                pre![&pre_class, arg_str],
+                if step.console.is_empty() {
+                    vec![]
+                } else {
+                    vec![
+                        h4![class![C.text_lg, C.font_medium], "Logs"],
+                        pre![&pre_class, step.console],
+                    ]
+                }
+            ]
+        ]
+    };
+    vec![item_caption, item_body]
+}
+
+fn status_text(cmd: &RichCommand) -> &'static str {
+    if cmd.complete {
+        "Complete"
     } else if cmd.errored {
         "Errored"
-    } else if cmd.complete {
-        "Complete"
+    } else if cmd.cancelled {
+        "Cancelled"
     } else {
         "Running"
     }
 }
 
-fn status_icon<T>(cmd: &Command) -> Node<T> {
-    let cls = class![C.w_4, C.h_4, C.inline, C.mr_4];
-
-    if cmd.cancelled {
-        font_awesome(cls, "ban").merge_attrs(class![C.text_gray_500])
+fn cmd_status_icon<T>(cmd: &RichCommand) -> Node<T> {
+    let awesome_class = class![C.w_4, C.h_4, C.inline, C.mr_4];
+    if cmd.complete {
```

Comment:
The ordering here needs to be reverted, complete *must* come last in the order.

Reply:
Ok, thanks.